pkg/linux/bandwidth: don't start bandwidth manager in dry mode

dylandreimerink opened the pull request:
When dry mode is enabled, we should not register the start hook. The start hook will probe and initialize the bandwidth manager, which should not happen in dry mode.

This was supposed to go into https://github.com/cilium/cilium/pull/28619 but due to a mistake it got merged without this change. So fixing that with this followup PR.

```release-note
do not start bandwidth manager in dry mode
```

dylandreimerink committed:
pkg/linux/bandwidth: don't start bandwidth manager in dry mode

When dry mode is enabled, we should not register the start hook. The
start hook will probe and initialize the bandwidth manager, which should
not happen in dry mode.

Signed-off-by: Dylan Reimerink <dylan.reimerink@isovalent.com>

marseel reviewed (approved):
Thanks, looks good from my side.

dylandreimerink committed:
pkg/datapath: Add bandwidth manager to datapath parameters

As a result of a bad merge in the original PR, the bandwidth manager
was not added to the params of the datapath cell.

Signed-off-by: Dylan Reimerink <dylan.reimerink@isovalent.com>
